src/Weather/Skysight/SkySightRequest.cpp

```
@@ -0,0 +1,996 @@
+    if (i->second->kind != FileJob::Kind::Generic) {
+      ++i;
+      continue;
+    }
+
+    i->second->function.Cancel();
+    retry_after.erase(i->first);
+    i = file_jobs.erase(i);
+  }
+
+  tile_http_error_count.clear();
+  PumpQueue();
```

Cppcheck (2.21.0)

[error] Unhandled exception thrown in function declared not to throw exceptions.

(throwInNoexceptFunction)